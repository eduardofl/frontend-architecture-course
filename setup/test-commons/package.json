{
  "name": "@eduardofl/test-commons",
  "version": "1.0.0",
  "private": true,
  "main": "index.js",
  "license": "MIT",
  "files": [
    "base.ts"
  ],
  "devDependencies": {
    "jest": "27.0.0",
    "ts-node": "10.7.0",
    "babel-jest": "28.0.3",
    "@babel/core": "7.17.10",
    "@babel/preset-env": "7.17.10",
    "@babel/preset-react": "7.16.7",
    "@babel/preset-typescript": "7.16.7",
    "@testing-library/react": "14.0.0",
    "jest-styled-components": "7.1.1"
  },
  "scripts": {
      "test": "echo \"Not implemented\"",
      "lint": "echo \"Not implemented\""
  }
}
