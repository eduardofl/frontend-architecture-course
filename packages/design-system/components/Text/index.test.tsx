import { Text } from "./index";
import { customRender } from "@eduardofl/test-commons/react-testing-library";

const render = customRender();

describe("<Text />", () => {
  it("should render h1 tag", () => {
    const { container } = render(<Text tag="h1">Hello World!</Text>);

    expect(container).toMatchSnapshot();
  });
});
