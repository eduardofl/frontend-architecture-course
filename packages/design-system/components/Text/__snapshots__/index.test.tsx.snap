// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should render h1 tag 1`] = `
.c0 {
  font-family: sans-serif;
}

<div>
  <h1
    class="c0"
  >
    Hello World!
  </h1>
</div>
`;
